src/dirichlet_multi.jl

```diff
@@ -26,8 +26,7 @@ posterior_canon(pri::Dirichlet, ss::MultinomialStats) = DirichletCanon(pri.alpha
 function posterior_canon(pri::Dirichlet, G::Type{Multinomial}, x::Matrix{T}) where T<:Real
 	d = length(pri)
 	size(x,1) == d || throw(ArgumentError("Inconsistent argument dimensions."))
-	a = add!(sum(x, dims=2), pri.alpha)
-	DirichletCanon(vec(a))
+	DirichletCanon(vec(sum(x, dims=2)) + pri.alpha )
```

Comment:
This causes more allocations than the old code.
```suggestion
        y = vec(sum(x; dims=2))
        y .+= pri.alpha
	DirichletCanon(y)
```